sklearn/preprocessing/_discretization.py

```diff
@@ -244,6 +274,22 @@ def transform(self, X):
         Xt : numeric array-like or sparse matrix
             Data in the binned space.
         """
+        X = check_array(X, dtype='numeric')
```

Comment:
If you need a `check_array` here, you should remove the next one `Xt = check_array(X, copy=True, dtype=FLOAT_DTYPES)`. 
You should also use the same parameters.